src/java.desktop/macosx/classes/sun/awt/CGraphicsEnvironment.java

```diff
@@ -139,13 +143,21 @@ void _displayReconfiguration(int displayId, boolean removed) {
         rebuildDevices();
     }
 
-    @Override
-    @SuppressWarnings("removal")
-    protected void finalize() throws Throwable {
-        try {
-            super.finalize();
-        } finally {
-            deregisterDisplayReconfiguration(displayReconfigContext);
+    private static class CGEDisposerRecord implements DisposerRecord {
+        private long displayReconfigContext;
+
+        CGEDisposerRecord(long ptr) {
+            displayReconfigContext = ptr;
+        }
+
+        public synchronized void dispose() {
+            if (displayReconfigContext != 0L) {
```

Comment:
can it be 0 here? it seems we never register 0 pointer, and we have a guard in native to be safe.

Reply:
and is there any harm in the check ?

Reply:
I removed the check